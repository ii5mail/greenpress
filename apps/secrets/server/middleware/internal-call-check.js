const { internalSecret } = require('../../config')

module.exports = (req, res, next) => {
  if (req.headers.internal_secret === internalSecret) {
    next()
  } else {
    res.status(401).json({ message: 'you are not allowed' }).end()
  }
}
