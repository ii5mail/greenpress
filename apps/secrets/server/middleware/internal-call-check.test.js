describe('internal-call-check', () => {

  let config;
  beforeEach(() => {
    jest.mock('../../config');
    config = require('../../config');
  });

  afterEach(() => {
    jest.unmock('../../config');
    jest.clearAllMocks();
  });

  describe('when request includes valid secret', () => {
    test('should continue with express middlewares', () => {
      const next = jest.fn();
      const req = { headers: { internal_secret: config.internalSecret } };

      require('./internal-call-check')(req, {}, next);

      expect(next).toBeCalled();
    });
  });

  describe('when request not includes valid secret', () => {
    test('should response not authorized to user', () => {

      config.internalSecret = 'aaaa';

      const req = { headers: { internal_secret: 'bbbb' } };

      const res = {
        status: jest.fn().mockImplementation(() => res),
        json: jest.fn().mockImplementation(() => res),
        end: jest.fn().mockImplementation(() => res),
      };

      require('./internal-call-check')(req, res);

      expect(res.status).toBeCalledTimes(1);
      expect(res.status).toBeCalledWith(401);

      expect(res.json).toBeCalledTimes(1);
      expect(res.json).toBeCalledWith({ 'message': 'you are not allowed' });

      expect(res.end).toBeCalledTimes(1);

      expect({
        status: res.status.mock.calls,
        json: res.json.mock.calls,
        end: res.end.mock.calls,
      }).toMatchSnapshot();
    });

  });

})
