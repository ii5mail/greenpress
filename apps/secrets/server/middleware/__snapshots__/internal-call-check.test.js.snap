// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`internal-call-check when request not includes valid secret should response not authorized to user 1`] = `
Object {
  "end": Array [
    Array [],
  ],
  "json": Array [
    Array [
      Object {
        "message": "you are not allowed",
      },
    ],
  ],
  "status": Array [
    Array [
      401,
    ],
  ],
}
`;
